tripal/src/Form/TripalImporterForm.php

```diff
@@ -0,0 +1,570 @@
+    // Quick check to make sure we had an array returned so array_merge() works.
+    if (!is_array($element_form)) {
+      $element_form = arry();
+    }
+
+    // Merge the custom form with our default one.
+    // This way, the custom TripalImporter can use the #weight property
+    // to change the order of their elements in reference to the default ones.
+    $form = array_merge($form, $element_form);
+
+    $form['button'] = [
+      '#type' => 'submit',
+      '#value' => t($class::$button_text),
+      '#weight' => 10,
+    ];
+    return $form;
+  }
+
+  /**
+   * Validate function for the tripal_get_importer_form form().
+   */
+  function tripal_get_importer_form_validate($form, &$form_state) {
+
+    $class = $form_state['values']['importer_class'];
+    tripal_load_include_importer_class($class);
+
+    $file_local = NULL;
+    $file_upload = NULL;
+    $file_remote = NULL;
+    $file_existing = NULL;
+
+    // Get the form values for the file.
+    if (array_key_exists('file_local', $class::$methods) and $class::$methods['file_local'] == TRUE) {
+      $file_local = trim($form_state['values']['file_local']);
+      // If the file is local make sure it exists on the local filesystem.
+      if ($file_local) {
+        // check to see if the file is located local to Drupal
+        $file_local = trim($file_local);
+        $dfile = $_SERVER['DOCUMENT_ROOT'] . base_path() . $file_local;
+        if (!file_exists($dfile)) {
+          // if not local to Drupal, the file must be someplace else, just use
+          // the full path provided
+          $dfile = $file_local;
+        }
+        if (!file_exists($dfile)) {
+          form_set_error('file_local', t("Cannot find the file on the system. Check that the file exists or that the web server has permissions to read the file."));
+        }
+      }
+    }
+    if (array_key_exists('file_upload', $class::$methods) and $class::$methods['file_upload'] == TRUE) {
+      $file_upload = trim($form_state['values']['file_upload']);
+      if (array_key_exists('file_upload_existing', $form_state['values']) and $form_state['values']['file_upload_existing']) {
+        $file_existing = $form_state['values']['file_upload_existing'];
+      }
+    }
+    if (array_key_exists('file_remote', $class::$methods) and $class::$methods['file_remote'] == TRUE) {
+      $file_remote = trim($form_state['values']['file_remote']);
+    }
+
+    // The user must provide at least an uploaded file or a local file path.
+    if ($class::$file_required == TRUE and !$file_upload and !$file_local and !$file_remote and !$file_existing) {
+      form_set_error('file_local', t("You must provide a file."));
+    }
+
+    // Now allow the loader to do validation of it's form additions.
+    $importer = new $class();
+    $importer->formValidate($form, $form_state);
+  }
+
+  /**
+   * Submit function for the tripal_get_importer_form form().
+   */
+  function tripal_get_importer_form_submit($form, &$form_state) {

```

Comment:
Are these the old D7 versions? If so, they should be removed entirely at this point.